packages/use-i18n/src/index.js

```diff
@@ -114,15 +130,22 @@ const I18nContextProvider = ({
       const { prefix, ...values } = trad
       const preparedValues = prefix ? prefixKeys(`${prefix}.`)(values) : values
 
-      setTranslations(prevState => ({
-        ...prevState,
-        ...{
-          [currentLocale]: {
-            ...prevState[currentLocale],
-            ...preparedValues,
+      // avoid a lot of render when async update
+      ReactDOM.unstable_batchedUpdates(() => {
```

Comment:
any idea why is it "unstable"?

Reply:
Ok got it in your article

Although this function is supposedly “unstable”, React apparently intends to address this in the following versions.

    “In future versions (probably React 17 and later), React will batch all updates by default so you won’t have to think about this”, according to Dan Abramov.

Reply:
It's unstable for the moment but it's use in React, Redux library from a long time. 
In a future upgrade of react, i thinks it's will not necessary